### ISSUE TITLE
[Propagation] Finish complete Tests of Propogation

### ISSUE THREAD

**Issue by ryanbanderson** [USGS-Astrogeology/PySAT_Point_Spectra_GUI#213]
Updating the data set name field does not propagate to Regression train and predict. In this screenshot, I have just changed the data set name and clicked "Refresh Modules":
![image](https://user-images.githubusercontent.com/3344800/36568444-47afedf0-17e7-11e8-8ae9-775cadd54d66.png)

**Comment by ryanbanderson** [comment]
Program crashes if the Refresh button is pressed and Regression train has not selected an algorithm.

**Comment by ryanbanderson** [comment]
The data keys created by stratified folds do not propagate.

**Comment by tisaconundrum2** [comment]
```
RegressionTrain.py:39: sklearn.exceptions.NotFittedError: This PLSRegression instance is not fitted yet. Call 'fit' with appropriate arguments before using this method.
```

Part of why Regression Train is not adding the new prediction to the dataframe has to do with the fact that we need to run `fit` 

Please advise on what we should do with this

**Comment by tisaconundrum2** [comment]
The current problem being dealt with is having items in ComboBoxes changing correctly. 
To highlight this issue I've included a small example below.

```
class Module:
    some_list = []

    def setList(self, _list, _item):
        try:
            _list[_list.index(_item)] = _item
        except:
            _list.append(_item)


class someUI(Module):
    index = 0
    def __init__(self):
        super().__init__()

    def setList(self, _list, _item):
        super().setList(_list, _item)


f = someUI()
g = someUI()
f.setList(f.some_list, 'item1')
g.setList(g.some_list, 'item2')
```
As you can see I've added a list, this list is globally accessible to any classes that extend the `Module` class. 
The expected result should be that every time we add new Modules (given by `f` and `g`) their corresponding space in `some_list` should be remembered. 

In other words, when a new module is added a new space should be added into `some_list`, whenever that module is edited we want only that item in `some_list` to changed. 
For example: If I'm messing with `f` then the corresponding location `some_list[0]` should be the only place things change. If I'm messing with `g` then the corresponding location `some_list[1]` should be the only place things change.  As you can see `setList` is an attempt at doing this.

This should remain consistent as we add on new items and classes.

```
class anotherUI(Module):
     ...

h = anotherUI()
h.setList(h.some_list, 'item3')
```

If you have any suggestions on how this can be better approached, I'm all ears.

**Comment by ryanbanderson** [comment]
This sounds close to what I was imagining, but in your example, don't you end up with different "some_list" lists for each module? Seems like we would want the list to exist at a higher level so that there's a single list and adding modules adds entries in it. We can talk about this in person today if you're in.
